pkg/kubectl/cmd/create.go

```diff
@@ -132,7 +133,12 @@ func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *C
 			}
 		}
 
-		if err := createAndRefresh(info); err != nil {
+		client, err := f.DynamicClient(info.Mapping.GroupVersionKind.GroupVersion())
```

Comment:
Unfortunately it is this way because of the underlying RESTClient structure and GroupVersions being closely tied to the codec at that level. This association doesn't matter for dynamic client, because it uses the same codec no matter what, but it's still a limitation of the RESTClient.

We'd have to fix that first.